test/integration/agent/net_windows_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agent
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/util"
+)
+
+func adapterName(name string) string {
+	return fmt.Sprintf("vEthernet (%s)", name)
+}
+
+func skipIfHyperVDisabled(t *testing.T) {
+	t.Logf("Checking if Hyper-V feature is enabled")
+	enabled, err := util.WindowsHyperVEnabled()
+	require.NoError(t, err)
+	if !enabled {
+		t.Skipf("Skipping test as it requires the Hyper-V feature to be enabled")
+	}
+}
+
+func skipIfMissingAdapter(t *testing.T, adapterName string) {
+	t.Logf("Checking if adapter '%s' exists", adapterName)
+	if _, err := net.InterfaceByName(adapterName); err != nil {
+		t.Skipf("Skipping test because we cannot verify that adapter '%s' exists", adapterName)
+	}
+}
+
+func skipIfOVSExtensionNotInstalled(t *testing.T) {
+	t.Logf("Checking if OVS vSwitch extension is installed")
+	cmd := `Get-VMSystemSwitchExtension -Name "Open vSwitch Extension"`
+	if err := util.InvokePSCommand(cmd); err != nil {
+		t.Skipf("Skipping test because we cannot verify that the OVS vSwitch extension is installed")
```

Comment:
Is "OVS vSwitch extension" correct? I see L52 uses "Open vSwitch Extension"

Reply:
I think I meant the OVS extension for the Windows vSwitch, but can update the logs to use the actual name of the extension - "Open vSwitch Extension" - everywhere.